### ISSUE TITLE
Should bins implicitly filter data that is out of range?

### ISSUE THREAD

**Issue by jakevdp** [vega/vega#2227]
Vega-lite's default for binning is to assign out-of-range data to the nearest defined bin. This is not a default I've come across in other charting libraries, and is potentially *very* misleading. For example, here is a zoomed-in normal distribution ([vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4zEgZwKYEcCuWAdgMZYpqaIBO85ADADRQbwD2ADuQIx28C+fBuCjwqsQhgBmrKgFtyAbXTFYAG2J5VCMqkyxZ7VVgByM2WoAUASkhNIsDOUgAPSHwC6QiJHNUA1k4ARrBUtsKQRMSs0ACWhADm5KAQ3q6o6PAAnuw6UPji8DGIhQBuZHaBcUkRzvBEtKgKALRcTFzuglCSMViq0E6ugsLemdVZOU75hIXFMWW2ULDx8VRY8dpOUXjTbsJ8IHxAA)):
```json
{
  "data": {"sequence": {"start": 0, "stop": 1000}},
  "transform": [{"calculate": "sampleNormal()", "as": "x"}],
  "mark": "bar",
  "encoding": {
    "x": {"field": "x", "type": "quantitative", "bin": {"extent": [-1, 1]}},
    "y": {"type": "quantitative", "aggregate": "count"}
  }
}
```
![visualization (57)](https://user-images.githubusercontent.com/781659/69984579-46c04880-14ee-11ea-8444-052f86ebacf2.png)

If you don't know to look for this (and my guess is most users wouldn't) it would lead you to make erroneous conclusions about the content of the data.

**Comment by jheer** [comment]
I like the proposal to output either null or infinite values for out-of-range bins. I will look into that soon!

**Comment by jheer** [comment]
Please see #2228 for a proposed solution.

**Comment by zanarmstrong** [comment]
thank you!

**Comment by jheer** [comment]
Fixed by #2228, will be included in v5.9.0.
